Library/Homebrew/os/linux/elf.rb

```diff
@@ -68,6 +68,40 @@ def binary_executable?
     elf_type == :executable
   end
 
+  def rpath
+    return @rpath if defined? @rpath
+
+    @rpath = if HOMEBREW_PATCHELF_RB
+      patcher = patchelf_patcher
+      begin
+        begin
+          patcher.runpath
```

Comment:
maybe we can have a tiny optimization of looking up `.rpath` before `.runpath`.
Not doing it, to behave like `patchelf` does, `.runpath` > `.rpath`

Reply:
We should check `runpath` first before `rpath`, for compatibility with the Linux kernel.

Reply:
Is `RUNPATH` before `RPATH` consistent with the dynamic linker's behavior? My understanding was that the dynamic linker searches `RPATH` before it searches `RUNPATH`, so a discovery in `RPATH` should always come first.

Reply:
The behaviour of the dynamic linker is a bit weird.
If `RUNPATH` exists, then first `LD_LIBRARY_PATH` is searched and then `RUNPATH` is searched.
If `RUNPATH` does not exist, then first `RPATH` is searched and then `LD_LIBRARY_PATH` is searched.
So `RUNPATH` takes precedence over `RPATH`, although their search order WRT `LD_LIBRARY_PATH` is different.

Search order:

1. `RPATH` so long as `RUNPATH` does _not_ exist
2. `LD_LIBRARY_PATH`
3. `RUNPATH`
4. … other stuff